src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledPackageFeed.cs

```diff
@@ -61,6 +62,20 @@ internal async Task<IPackageSearchMetadata[]> GetMetadataForPackagesAndSortAsync
             return items.OrderBy(p => p.Identity.Id).ToArray();
         }
 
+        internal static async Task<IPackageSearchMetadata[]> GetMetadataFromIdentityForPackagesAndSortAsync<T>(T[] packages, bool includePrerelease, CancellationToken cancellationToken) where T : PackageIdentity
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            IEnumerable<IPackageSearchMetadata> items = await TaskCombinators.ThrottledAsync(
+                packages,
+                (p, t) => Task.FromResult(PackageSearchMetadataBuilder.FromIdentity(p).Build()),
+                cancellationToken);
+
+            // The packages were originally sorted which is important because we skip based on that sort
+            // however, the asynchronous execution has randomly reordered the set. So, we need to resort.
+            return items.OrderBy(p => p.Identity.Id).ToArray();
```

Comment:
How do we ensure this sort stays consistent with the original sorting? Can we call the same method to sort so we don't duplicate code and risk them being out of sync should they be modified later?

Reply:
Moved this to another method and removed the "AndSort" from these functions